doc/source/serve/handle-guide.md

```diff
@@ -2,74 +2,59 @@
+
+This is particularly useful for two use cases:
+Calling deployments dynamically within the deployment graph.
+Iterating and testing your application in Python.
+
+To use the ServeHandle, use {mod}`handle.remote <ray.serve.handle.RayServeHandle.remote>` to send requests to that deployment.
+These requests can pass ordinary args and kwargs that are passed directly to the method. This returns a Ray `ObjectRef` whose result can be waited for or retrieved using `await` or `ray.get`.
+
+Conceptually, ServeHandle is a client side load balancer. It has the ability to route requests to any replicas of a given deployment. It also performs buffering internally so it won’t overwhelm the replicas. The number of requests buffered is used to inform autoscaler to scale up the number of requests.
 
-If you want to use the same deployment to serve both HTTP and ServeHandle traffic, the recommended best practice is to define an internal method that the HTTP handling logic will call:
+![architecture-diagram-of-serve-handle](https://raw.githubusercontent.com/ray-project/images/master/docs/serve/serve-handle-explainer.png)
 
-```python
-@serve.deployment(route_prefix="/api")
-class Deployment:
-    def say_hello(self, name: str):
-        return f"Hello {name}!"
+ServeHandle takes request parameters and returns a `ray.ObjectRef`. The `ray.ObjectRef` corresponds to a future object that will be fulfilled with the result object. Because of the internal buffering, the time from submitting a request to getting a `ray.ObjectRef` varies from instantaneous to indefinitely long.
 
-    def __call__(self, request):
-        return self.say_hello(request.query_params["name"])
+Because of this, we have two types of handle to make sure the buffering period is handled efficiently:
+- `RayServeDeploymentHandle` returns an `asyncio.Task` upon submission. The `asyncio.Task` can be awaited to resolve to a ray.ObjectRef. When a single request is being buffered, other requests can be processed concurrently.
+- `RayServeSyncHandle` directly returns a `ray.ObjectRef`. It blocks the current thread until the request is matched to a replica. This also matches the Ray Core Actor API.
 
-Deployment.deploy()
+When you call `serve.run` to deploy a deployment graph (including single node graph), the driver node’s handle is returned. The return type is a ServeSyncHandle. This is useful for interacting with the deployment graph you just created and test against it.
+
+```{literalinclude} ../serve/doc_code/handle_guide.py
+:start-after: __begin_sync_handle__
+:end-before: __end_sync_handle__
+:language: python
 ```
 
-Now we can invoke the same logic from both HTTP or Python:
+Other than `serve.run`, RayServeDeploymentHandle is the default because the API is more performant than its blocking counterpart. For example, when implementing a dynamic dispatch node in deployment graph, the handle is asynchronous.
```

Comment:
```suggestion
In all other cases, `RayServeDeploymentHandle` is the default because the API is more performant than its blocking counterpart. For example, when implementing a dynamic dispatch node in deployment graph, the handle is asynchronous.
```